mne/io/eeglab/eeglab.py

```diff
@@ -355,6 +351,22 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         _read_segments_file(self, data, idx, fi, start, stop, cals, mult,
                             dtype=np.float32, n_channels=self.info['nchan'])
 
+    # XXX: to be removed when deprecating montage
```

Comment:
So this PR still does not do what we want to do, which is just to call `self.set_montage` at the end of init?

Reply:
Once `set_montage` uses Digitization there would be no need for all this monkeypatching in EEGLAB (which is the only place where weird things are there)

the `__init__` does call self.set_montage(montage)